### PR TITLE
Submission/sprint6

### PR DESCRIPTION
**Pull request by PapaOscar90**

### CODE REVIEW

File: tests/modules/servers/resources/test_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    ('GET', '/servers/'),
+    ('POST', '/servers/'),
+    ('GET', '/servers/1'),
+    ('PUT', '/servers/1'),
+    ('DELETE', '/servers/1'),
+    ('POST', '/servers/1/ping'),
+    ('POST', '/servers/1/batch_request'),
+))
+
+def test_general_return(http_method, http_path, flask_app_client):
```
Comment (Feikoritsema) [inline tests/modules/servers/resources/test_endpoints.py]: All the tests "assert response is not None" is way too broad.

Nicer would be checking how many servers there are, then adding one and seeing if there is an extra server for example.

File: tests/modules/servers/resources/test_endpoints.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+"""
```
Comment (Feikoritsema) [inline tests/modules/servers/resources/test_endpoints.py]: Nice that you added some tests.

File: tests/modules/users/test_users.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+"""
```
Comment (Feikoritsema) [inline tests/modules/users/test_users.py]: If you don't create the tests, it's useless to add this

File: tests/test_dummy.py [hunk 4]
```diff
@@ -1,3 +1,7 @@
-import sqlalchemy
+"""
+Dummy test to check Pytest and CI functions.
+"""
```
Comment (Feikoritsema) [inline tests/test_dummy.py]: Dummy test can be deleted if you have actual tests.

File: app/modules/servers/resources.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        required=False,
+        description='The information to be forwarded to the remote server.'
+    ),
+})
+
+
+@NAMESPACE.route('/<string:server_id>/request')
+@NAMESPACE.param('server_id', 'The server identifier')
+class ServerRequest(Resource):
+    """
```
Comment (Feikoritsema) [inline app/modules/servers/resources.py]: Nice docstring

File: app/extensions/flask_sqlalchemy/__init__.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+"""
+import sqlite3
+
+from sqlalchemy import engine, MetaData
+
+from flask_sqlalchemy import SQLAlchemy as BaseSQLAlchemy
+
+
+def set_sqlite_pragma(dbapi_connection, connection_record):
+    # pylint: disable=unused-argument
```
Comment (Feikoritsema) [inline app/extensions/flask_sqlalchemy/__init__.py]: Why? Is connection_record needed ?

File: app/extensions/auth/authentication.py [hunk 7]
```diff
@@ -1,177 +1,189 @@
...
+            try:
+                payload = jwt.decode(
+                    token,
+                    rsa_key,
+                    algorithms=ALGORITHMS,
+                    audience=API_IDENTIFIER,
+                    issuer="https://"+AUTH0_DOMAIN+"/"
+                )
+            except jwt.ExpiredSignatureError:
+                raise AuthError({"code": "token_expired",
```
Comment (Feikoritsema) [inline app/extensions/auth/authentication.py]: Logging would be good too.

**Comment by PapaOscar90** [comment]
I apologize nobody has handled the comments and review yet. I will be able to tend to these this afternoon.

File: tests/modules/servers/resources/test_endpoints.py (hunk 1 above)
Comment (NikDijkema) [reply tests/modules/servers/resources/test_endpoints.py]: We are currently finishing fixing the tests, the main hold-up was in overwriting the authentication via Auth0, so the tests can be done locally in isolation. Will be fixed in the upcoming sprint request.

File: app/extensions/flask_sqlalchemy/__init__.py (hunk 6 above)
Comment (NikDijkema) [reply app/extensions/flask_sqlalchemy/__init__.py]: It is required by SQLAlchemy when it instatiates the DB connection.

File: docs/architecture/HestiaWeb_architecture.md [hunk 8]
```diff
@@ -1,89 +1,206 @@
...
+We initially decided to use [Google Firebase](https://firebase.google.com/) for login and storing each users' servers, however then we switched to [Auth0](https://auth0.com/) for login, in order to implement single sign-on. Once the user is logged in, the user's information will be used to get all their servers from our [PostgreSQl](https://www.postgresql.org/) database, and connect to these servers. The front-end will be developed in [Vue.js](https://vuejs.org/), in  order to simplify the development of our user interface. The main focus of the website will be on listing the devices managed by their controllers, and on the operations that can be applied to them. These will mirror those already implemented by Hestia, such as renaming or deleting a device, but with more streamlined interfacing added, such as using buttons and sliders in order to change the activators of a device, instead of having to directly enter values. Additionally, more functionality such as the ability to batch apply presets will be exposed in the back-end and interfaced via this front-end.
+
+Within Vue, three design patterns were used to simply development. They were:
```
[the remark this answers is not in the document]
Comment (NikDijkema) [reply docs/architecture/HestiaWeb_architecture.md]: We've updated the documentation, this i now resolved.

**Pull request merged**